lib/jekyll-relative-links/generator.rb

```diff
@@ -23,15 +26,19 @@ def initialize(site)
     def generate(site)
       @site    = site
       @context = context
+      return if disabled?
 
-      site.pages.each do |page|
+      pages = site.pages
+      pages = site.pages + site.documents if collections?
+
+      pages.each do |page|
         next unless markdown_extension?(page.extname)
         replace_relative_links!(page)
       end
     end
 
     def replace_relative_links!(page)
-      url_base = File.dirname(page.path)
+      url_base = File.dirname(page.relative_path)
```

Comment:
I ran into some strange behavior with this. https://github.com/benbalter/jekyll-relative-links/pull/27/files#diff-1c96af68dcbaa3116d54508be6c5ca8bR140 was my work around to deal with the different behavior between pages and documents when it comes to paths.

Reply:
hm, on the surface it looks like it `path` and `relative_path` are always identical for pages. do have more details on where this behavior occurred?

Reply:
Read through the code and the above linked PR and I think I figured out the issue:

https://github.com/benbalter/jekyll-relative-links/blob/36318fd8699013a538823c86b8da9340c84dad77/lib/jekyll-relative-links/generator.rb#L34 is intended to set `url_base` as the path to the linking file's parent directory relative to the site source. That's then passed to [`path_from_root`](https://github.com/benbalter/jekyll-relative-links/blob/36318fd8699013a538823c86b8da9340c84dad77/lib/jekyll-relative-links/generator.rb#L40) as the URL base along with the relative path to the link to determine the linked file's path relative to the site source.

I think the issue here may be that Pages and Documents would behave differently when the site's source was not the working directory (or you may not be able to link from a page to a document) since the paths would not match.